xds/internal/client/xds.go

```diff
@@ -522,7 +522,7 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 				route.WeightedClusters[c.GetName()] = wc
 				totalWeight += w
 			}
-			if totalWeight != wcs.GetTotalWeight().GetValue() {
+			if totalWeight != 100 && (wcs.GetTotalWeight() == nil || totalWeight != wcs.GetTotalWeight().GetValue()) {
```

Comment:
This should work. But let's change it a little bit for readability, and more accurate error message

```go
wantTotalWeight := 100
if tw := wcs.GetTotalWeight(); tw != nil {
  wantTotalWeight = tw.GetValue()
}
if totalWeight != wantTotalWeight {
  return nil, fmt.Errorf("route %+v, action %+v, weights of clusters do not add up to total total weight, got: %v, expected total weight from response: %v", r, a, totalWeight, wantTotalWeight)
}
```

Reply:
thx, resolved